### ISSUE TITLE
ServerApp 100% crashes when connecting to client on Android

### ISSUE THREAD

**Issue by HAL-XP** [sammyfreg/netImgui#24]
First cmd packets coming from the clients are ok (Ping, Background) but the first DrawFrame makes the app crash on read access violation. Inside void CmdDrawFrame::ToPointers() when checking if( !mpDrawGroups.IsPointer() ) the data has the right signature and looks ok (0x800000something) but crashes on first (i = 0) mpDrawGroups[i].ToPointers(); 

I guess we hit the limits of the following comment :)
// @sammyfreg TODO: Make Offset/Pointer test safer

**To Reproduce**
1. Launch Client (default ports) on Android
2. Launch Server
3. See error

![netimgui_server_crash](https://user-images.githubusercontent.com/30446630/153608393-d775a1c8-c4c3-46ba-bad2-5dd971e62937.png)

**Desktop (please complete the following information):**
 - OS: Win64

**Smartphone (please complete the following information):**
 - Device: Google Pixel 6 Pro
 - OS: [latest]

**Comment by sammyfreg** [comment]
I am not setuped to run on Android :/

But first thing first, is this behaviour occuring with compression off?
(Either set the client to disable it with NetImguiApi, or turn it off in
the editor settings)

**Comment by sammyfreg** [comment]
I do not believe this is the cause, but could you test with latest Dev version ([3612a56](https://github.com/sammyfreg/netImgui/commit/3612a56e5ad2e113bd15bc46872f4b6f6c2eb144)) ? I fixed some issues 32bits clients and with crash when there's no Dear ImGui draws in a DrawFrameCmd.

**Comment by HAL-XP** [comment]
Sure! Will try tonight, thanks!

**Comment by sammyfreg** [comment]
Nice!

My first suspicion was an empty drawlist but when I tried generating one on PC, it crashed on the Client before it could be sent to the Server. Interesting that on Android it was able to access Vector[0] with no element allocated.

**Comment by coodeneko** [comment]
Sorry to resurect a closed issue but I am experincing the exactly same issue when using an arm64 binary. armv7 (32 bit) is okay.
I also tried turning compression off.
I tried in both the latest in master and the latest in dev ([6010086](https://github.com/sammyfreg/netImgui/commit/6010086d70504f537a0506f13b99a991ca393a3f))

**Comment by sammyfreg** [comment]
That's peculiar. Your server runs on Windows and your Android 32bits client
works but but the 64bits ?

Could it be tied to int endian ordering? I did not code any handling for it
and it assumes Intel ordering.

**Comment by coodeneko** [comment]
Wow, thanks for replying so quickly.
I did a quick test and it's reporting that its operating in Little Endian mode.

I'm also using the Unreal plugin https://github.com/sammyfreg/UnrealNetImgui
Maybe that has something to do with it.
Going on new years break right away, if I find out something more I'll report back after that.

**Comment by coodeneko** [comment]
The bug in my situation was the address space on my side was high enough that the bit used for marking offsets was set. Likely only an issue with how our setup is running. I have a local hack that let's us use this great tool.

**Comment by sammyfreg** [comment]
Ah... This was always an assert optimisation that had the potential for
trouble like you just encountered.

I might have to revise this without the assumption of the top bit always
being unused for addresses.

**Comment by sammyfreg** [comment]
I updated the Dev branch with a fix, let me know if it takes care of it (change d882f57e89b30df1b283e4091ddb931749519854).
